de.metas.swat/de.metas.swat.base/src/main/java/de/metas/inoutcandidate/api/impl/ShipmentSchedulePA.java

```diff
@@ -1530,12 +1530,7 @@ else if (tableID == InterfaceWrapperHelper.getTableId(I_M_InOutLine.class))
 
 			schedules = ImmutableSet.copyOf(retrieveForInOutLine(inoutLine));
 		}
-		else
```

Comment:
usually it's not a good practice to have if/else if/... without an else. You can assign the schedules = ImmutableSet.of() there, keep the comment/documentation  and remove the Check.errorIf.

Also when possible, favor using final variables, even locally.

Reply:
hm.. ok
But that was the idea with having Set<I_M_ShipmentSchedule> schedules; initialized with ImmutableSet.of()

Reply:
not related.
The advantage of having not initialized final variables is that you will get a compile error if you would forget to handle a case.
No, in case of any mistake like a missing if/else or something your schedules will be empty.
ping me if u don't get what i am talking about.

Reply:
ok. I got it